### PR TITLE
LPS-170077 Add new expire option to KB settings

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
- LPS-170077 Create a configuration Screen to add a proper style to the configuration of the KB articles
- LPS-170077 add new configuration for the expirationDateNotification and let the configuration screen be the one to paint them
- LPS-170077 only show the configuration if the FF is active
- LPS-170077 buildService
- LPS-170077 Add Service Configuration Provider for obtain the configuration of the KBArticle by company
- LPS-170077 baseline
- LPS-170077 add context and jsp
- LPS-170077 add impl of the provider and action command that call it
- LPS-170077 buildLang

image

The only discrepancy with the  Figma model is that the scope is the company

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
I'll rebase the PR due to he conflicts

**Review by marcogalluzzi — APPROVED** [review]
Nice!

**Comment by marcogalluzzi** [comment]
BTW I've you seen that in your screenshot the apostrophe character is not shown correctly. Is that normal?

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
a minor error > rebase

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by marcogalluzzi — APPROVED** [review]

**Comment by marcogalluzzi** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/configuration/KBServiceConfiguration.java [hunk 1]
```diff
@@ -35,4 +35,10 @@ public interface KBServiceConfiguration {
 	@Meta.AD(deflt = "15", name = "check-interval", required = false)
 	public int checkInterval();
 
+	@Meta.AD(
+		deflt = "1", description = "expiration-date-notification-help",
+		name = "expiration-date-notification", required = false
+	)
+	public int expirationDateNotification();
+
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/configuration/KBServiceConfiguration.java:43]: The method name is a bit unclear about the value it returns. I think we should add something more to it (e.g. `expirationDateNotificationWeekInterval()`.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by AliciaGarciaGarcia** [comment]
name changes 
Changed the names according with Design (figma update pending)
image

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]
@AliciaGarciaGarcia I'm ok with the code, but I think the labels are not good. They don't provide enough information (e.g. it talks about dates, but the field is an integer; it is not clear what the value means). We need to talk with design about it.

**Comment by AliciaGarciaGarcia** [comment]
I will rebase, remade the story and resend with new messages

Thanks!

**Comment by AliciaGarciaGarcia** [comment]
image

**Comment by AliciaGarciaGarcia** [comment]
@boton 

Can you take a look to the input part, I want to be sure I'm not making anything wrong

Thanks!

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by boton — APPROVED** [review]

**Review by marcogalluzzi — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia There are some baseline conflicts. Feel free to forward when resolved.

**Comment by AliciaGarciaGarcia** [comment]
rebased

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Pull request closed without merging**